core/utils/poly-look/src/visualisations/charts/bubble-charts/bubbleCluster/bubbleCluster.js

```diff
@@ -20,8 +20,96 @@ const defaultOnClickFunction = () => {};
 const nodeLabelBoxPaddingX = 4,
   nodeLabelBoxPaddingY = 1,
   nodeLabelBoxBorderRadius = 2,
-  defaultNodeLabelBoxColor = "white",
-  defaultNodeLabelBoxOpacity = 0.7;
+  nodeLabelBoxColor = "white",
+  nodeLabelBoxOpacity = 0.7,
+  nodeLabelLineMargin = 2,
+  nodeLabelLineStrokeWidth = 2;
+
+const getBCR = (selection) => selection.node().getBoundingClientRect();
+
+function positionNodeLabelOnX({ text, rect, line, side = "right" }) {
+  const textClientRect = getBCR(text);
+  const factor = side == "right" ? 1 : -1;
+  text &&
+    text
+      .attr("x", (d) => (d.r + labelMargin) * factor)
+      .attr("y", textClientRect.height / 4)
+      .attr("text-anchor", side == "right" ? "start" : "end");
+
+  text &&
+    text.each((_, i, nodes) => {
+      const bBox = nodes[i].getBBox();
+      rect
+        .attr("x", bBox.x - nodeLabelBoxPaddingX)
+        .attr("y", bBox.y - nodeLabelBoxPaddingY);
+    });
+
+  line &&
+    line
+      .attr("y1", 0)
+      .attr("y2", 0)
+      .attr("x1", (d) => (d.r + nodeLabelLineMargin) * factor)
+      .attr(
+        "x2",
+        (d) =>
+          (d.r + labelMargin - nodeLabelLineMargin - nodeLabelBoxPaddingX) *
+          factor
+      );
+}
+
+function positionNodeLabelOnY({ text, rect, line, side = "top" }) {
+  const textClientRect = getBCR(text);
+  const isTop = side == "top";
+  const factor = isTop ? -1 : 1;
+  text &&
+    text
+      .attr("x", 0)
+      .attr(
+        "y",
+        (d) =>
+          (d.r + labelMargin + (!isTop && textClientRect.height / 2)) * factor
+      )
+      .attr("text-anchor", "middle");
+
+  text &&
```

Comment:
74 - 94 seem to be identical with 39 - 59. Could be worth making it a function called addLabelPane or addLabelPlaceholder?